src/Space/index.tsx

```diff
@@ -1,4 +1,31 @@
 import { Space as AntdSpace, SpaceProps as AntdSpaceProps } from 'antd';
+import React from 'react';
+import styled from 'styled-components';
 
-export const Space: typeof AntdSpace = AntdSpace;
-export type SpaceProps = AntdSpaceProps;
+import { PREFIX_CLS } from '../Provider';
+import { classNames } from '../classNames';
+
+export { SpaceSize } from 'antd/es/space';
+
+export type SpaceProps = AntdSpaceProps & {
+  block?: boolean;
+};
+
+const BLOCK_CLASS = `${PREFIX_CLS}-space-block`;
+
+const StyledSpace = styled(AntdSpace)<SpaceProps>`
+  &.${BLOCK_CLASS} {
+    display: flex;
+  }
+`;
```

Comment:
Was the extra class needed to stop prop forwarding?

```suggestion
function shouldForwardProp(prop: keyof SpaceProps) {
  return !['block'].includes(prop);
}

export const Space = styled(Space).withConfig<SpaceProps>({
  shouldForwardProp,
})`
  ${(props) => (props.block ? 'display: flex;' : '')}
`;
```

Reply:
Nice, haven't used it so far. I tried to use transient props at first, but didn't like the resulting code. This is better.

Change the style definition a bit so that the linter recognizes invalid styles.